addons/l10n_in/models/res_config_settings.py

```diff
@@ -33,3 +53,30 @@ def l10n_in_edi_buy_iap(self):
             'url': self.env["iap.account"].get_credits_url(service_name=IAP_SERVICE_NAME),
             'target': '_new'
         }
+
+    def set_values(self):
+        if self.country_code == 'IN':
+            old_value = str2bool(self.env['ir.config_parameter'].get_param('l10n_in.l10n_in_is_gst_registered'))
+            if old_value != self.l10n_in_is_gst_registered:
+                if old_value:
+                    raise UserError(_('Once GST is enabled, it cannot be disabled.'))
+                self._archive_u_r_taxes()
+                self._activate_gst_taxes()
+                # Set sale and purchase tax accounts when GST is enabled.
+                self.company_id.account_sale_tax_id = self.env['account.chart.template'].ref('sgst_sale_5').id
+                self.company_id.account_purchase_tax_id = self.env['account.chart.template'].ref('sgst_purchase_5').id
+        return super().set_values()
+
+    def _archive_u_r_taxes(self):
+        taxes = self.env['account.tax'].with_context(active_test=False).search([
+            ('company_id', '=', self.company_id.id),
+            ('name', 'ilike', '% U R'),
```

Comment:
Use reference (external_id) of taxes not name.
I would suggest to use `_xmlid_to_res_id` to fetch the list of record ids and browse them rather ref.